tests/triggers/test_text_triggers.py

```diff
@@ -0,0 +1,167 @@
+import pytest
+from ...gramhopper.triggers.text_triggers import _RegExpTrigger, \
+    _HasSubstringTrigger, _HasExactWordTrigger
+
+
+@pytest.mark.usefixtures('update')
+class TestRegexpTrigger:
+    def test_regexp_trigger(self, update):
+        email_pattern = r'^([a-z0-9_\.-]+)@([\da-z\.-]+)\.([a-z\.]{2,6})$'
+        trigger = _RegExpTrigger(email_pattern)
+
+        # Assuring that the regexp matches the message text
+        update.message.text = 'user@example.com'
+        result = trigger.check_trigger(update)
+        assert result.should_respond
+        assert 'match' in result.response_payload
+        match = result.response_payload['match']
+        assert len(match) == 3
+        assert match[0] == 'user'
+        assert match[1] == 'example'
+        assert match[2] == 'com'
+
```

Comment:
I think you should separate those two tests:
1. Assert that a real email activates the trigger
2. Asset that a non-email doesn't activate the trigger

That way you can easily distinguish false-negatives from false-positives.

https://techbeacon.com/app-dev-testing/5-ways-simplify-your-automated-test-cases
(2. Fail for one, and only one, reason)